code/modules/food_and_drinks/food/snacks_frozen.dm

```diff
@@ -188,3 +191,73 @@
+	list_reagents = list(/datum/reagent/consumable/nutriment = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 2, /datum/reagent/consumable/sugar = 4)
+	tastes = list("beetlejuice")
+	trash = /obj/item/popsicle_stick
+	w_class = WEIGHT_CLASS_SMALL
+	var/overlay_state = "creamsicle_o" //This is the edible part of the popsicle.
+	var/bite_states = 4 //This value value is used for correctly setting the bitesize to ensure every bite changes the sprite. Do not set to zero.
+
+/obj/item/reagent_containers/food/snacks/popsicle/Initialize()
+	. = ..()
+	bitesize = reagents.total_volume / bite_states
+
+/obj/item/reagent_containers/food/snacks/popsicle/update_overlays()
+	. = ..()
+	if(bitecount)
+		. += "[initial(overlay_state)]_[min(bitecount, 3)]"
+	else
+		. += initial(overlay_state)
+
+/obj/item/reagent_containers/food/snacks/popsicle/On_Consume(mob/living/eater)
+	. = ..()
+	update_icon()
+
+/obj/item/popsicle_stick
+	name = "popsicle stick"
+	icon = 'icons/obj/food/frozen_treats.dmi'
+	icon_state = "popsicle_stick"
+	desc = "This humble little stick usually carries a frozen treat, at the moment it seems freed from this atlassian burden."
+	custom_materials = list(/datum/material/wood=20)
+	w_class = WEIGHT_CLASS_TINY
+	force = 0
+
+/obj/item/reagent_containers/food/snacks/popsicle/creamsicle_orange
+	name = "orange creamsicle"
+	desc = "A classic orange creamsicle. A sunny frozen treat."
+	list_reagents = list(/datum/reagent/consumable/orangejuice = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 2, /datum/reagent/consumable/sugar = 4)
+
+/obj/item/reagent_containers/food/snacks/popsicle/creamsicle_berry
+	name = "berry creamsicle"
+	desc = "A vibrant berry creamsicle. A berry good frozen treat."
+	list_reagents = list(/datum/reagent/consumable/berryjuice = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 2, /datum/reagent/consumable/sugar = 4)
+	overlay_state = "creamsicle_m"
+
+/obj/item/reagent_containers/food/snacks/popsicle/jumbo
+	name = "jumbo icecream"
+	desc = "A luxurius icecream covered in rich chocolate. It seems smaller than you remember it being."
+	list_reagents = list(/datum/reagent/consumable/hot_coco = 4, /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 3, /datum/reagent/consumable/sugar = 4)
+	overlay_state = "jumbo"
+
+/obj/item/reagent_containers/food/snacks/popsicle/nogga_black
+	name = "nogga black"
+	desc = "A salty licorice icecream recently reintroduced due to all the records of the controversy being lost to time. Those who cannot remember the past are doomed to repeat it."
+	list_reagents = list(/datum/reagent/consumable/nutriment = 3, /datum/reagent/consumable/sodiumchloride = 1,  /datum/reagent/consumable/cream = 2, /datum/reagent/consumable/vanilla = 1, /datum/reagent/consumable/sugar = 4)
+	tastes = list("salty liquorice")
+	overlay_state = "nogga_black"
+
+/obj/item/reagent_containers/food/snacks/cornuto
```

Comment:
Dottò...
![cdss](https://user-images.githubusercontent.com/5155917/73122875-f4e5ed00-3f89-11ea-82fc-6e409186b9a6.jpg)

Reply:
is the name intentional or just a nut pun

Reply:
The name is very intentional. Very similar to the original name, but a funnier meaning  in Italian

Reply:
Festa del Cornuto holiday when